src/Brick/Widgets/Core.hs

```diff
@@ -992,32 +995,28 @@ cropBottomTo rows p =
 -- | When rendering the specified widget, also register a cursor
 -- positioning request using the specified name and location.
 showCursor :: n -> Location -> Widget n -> Widget n
-showCursor n cloc p =
-    Widget (hSize p) (vSize p) $ do
-      result <- render p
-      return $ result & cursorsL %~ (CursorLocation cloc (Just n) True:)
+showCursor n cloc p = Widget (hSize p) (vSize p) $ 
+    fmap (cursorsL %~ (CursorLocation cloc (Just n) True:)) (render p)
 
 -- | When rendering the specified widget, also register a cursor
 -- positioning request using the specified name and location.
 -- The cursor will only be positioned but not made visible.
 putCursor :: n -> Location -> Widget n -> Widget n
-putCursor n cloc p =
-    Widget (hSize p) (vSize p) $ do
-      result <- render p
-      return $ result & cursorsL %~ (CursorLocation cloc (Just n) False:)
+putCursor n cloc p = Widget (hSize p) (vSize p) $
+    fmap (cursorsL %~ (CursorLocation cloc (Just n) False:)) (render p)
```

Comment:
I prefer infix `fmap` (`<$>`) when possible. I'll merge and update this case, but I just wanted you to know for future patches. Thank you!